jni_bindings/src/lib.rs

```diff
@@ -73,159 +69,226 @@ pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_00024Comp
+    this: jlong,
+    // Raw pointer to Document
 ) -> jlong {
-    unsafe {
-        let java_result: &mut JavaResult =
-            jlong_to_pointer::<JavaResult>(j_result).as_mut().unwrap();
-        java_result.val
+    let this = this as *mut JavaResult;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "JavaResult::get_val called with `this` as null pointer",
+        )
+        .unwrap();
+
+        0
+    } else {
+        let result = &mut *this;
+        result.val as jlong
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_JavaResult_get_1error<'local>(
-    env: JNIEnv<'local>,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_JavaResult_get_1error<
+    'local,
+>(
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    j_result: jlong,
+    // non-null raw pointer to JavaResult
+    this: jlong,
 ) -> JString<'local> {
-    unsafe {
-        let java_result: &JavaResult = jlong_to_pointer::<JavaResult>(j_result).as_mut().unwrap();
-        from_std_string_jstring(&java_result.error_msg, env)
+    let this = this as *mut JavaResult;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "JavaResult::get_error called with `this` as null pointer",
+        )
+        .unwrap();
+
+        JObject::null().into()
+    } else {
+        let result = &*this;
+
+        env.new_string(result.error_msg.as_str())
+            .unwrap_or_else(|_| JObject::null().into())
     }
 }
 
-// takes dom
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_do_1to_1string<'local>(
-    env: JNIEnv<'local>,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Document_do_1to_1string<
+    'local,
+>(
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    dom: jlong,
+    // non-null raw pointer to Document
+    this: jlong,
 ) -> JString<'local> {
-    unsafe {
-        let dom: &dom::Document = jlong_to_pointer::<dom::Document>(dom).as_mut().unwrap();
-        from_std_string_jstring(dom.to_string(), env)
+    let this = this as *mut Document;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Document::do_to_string call with `this` as null pointer",
+        )
+        .unwrap();
+
+        JObject::null().into()
+    } else {
+        let doc = &*this;
+
+        env.new_string(doc.to_string())
+            .unwrap_or_else(|_| JObject::null().into())
     }
 }
 
 // Java side should ensure only u32 is passed as the node parameter
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_node_1to_1string<'local>(
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Document_node_1to_1string<
+    'local,
+>(
     mut env: JNIEnv<'local>,
     _: JClass<'local>,
+    // non-null raw pointer to Document
```

Comment:
The `non-null` can't be true or there would be no need for the `this.is_null()` check below.
```suggestion
    // raw pointer to Document
```